cpp/src/io/fst/dispatch_dfa.cuh

```diff
@@ -0,0 +1,456 @@
+  // TYPEDEFS
+  //------------------------------------------------------------------------------
+  using StateVectorCompositeOpT = VectorCompositeOp<MAX_NUM_STATES>;
+
+  //------------------------------------------------------------------------------
+  // MEMBER VARS
+  //------------------------------------------------------------------------------
+  void* d_temp_storage;
+  size_t& temp_storage_bytes;
+  DfaT dfa;
+  StateIndexT seed_state;
+  SymbolItT d_chars_in;
+  OffsetT num_chars;
+  TransducedOutItT transduced_out_it;
+  TransducedIndexOutItT transduced_out_idx_it;
+  TransducedCountOutItT d_num_transduced_out_it;
+  cudaStream_t stream;
+  int const ptx_version;
+
+  //------------------------------------------------------------------------------
+  // CONSTRUCTOR
+  //------------------------------------------------------------------------------
+  CUB_RUNTIME_FUNCTION __forceinline__ DispatchFSM(void* d_temp_storage,
+                                                   size_t& temp_storage_bytes,
+                                                   DfaT dfa,
+                                                   StateIndexT seed_state,
+                                                   SymbolItT d_chars_in,
+                                                   OffsetT num_chars,
+                                                   TransducedOutItT transduced_out_it,
+                                                   TransducedIndexOutItT transduced_out_idx_it,
+                                                   TransducedCountOutItT d_num_transduced_out_it,
+                                                   cudaStream_t stream,
+                                                   int ptx_version)
+    : d_temp_storage(d_temp_storage),
+      temp_storage_bytes(temp_storage_bytes),
+      dfa(dfa),
+      seed_state(seed_state),
+      d_chars_in(d_chars_in),
+      num_chars(num_chars),
+      transduced_out_it(transduced_out_it),
+      transduced_out_idx_it(transduced_out_idx_it),
+      d_num_transduced_out_it(d_num_transduced_out_it),
+      stream(stream),
+      ptx_version(ptx_version)
+  {
+  }
+
+  //------------------------------------------------------------------------------
+  // DISPATCH INTERFACE
+  //------------------------------------------------------------------------------
+  CUB_RUNTIME_FUNCTION __forceinline__ static cudaError_t Dispatch(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    DfaT dfa,
+    StateIndexT seed_state,
+    SymbolItT d_chars_in,
+    OffsetT num_chars,
+    TransducedOutItT transduced_out_it,
+    TransducedIndexOutItT transduced_out_idx_it,
+    TransducedCountOutItT d_num_transduced_out_it,
+    cudaStream_t stream)
+  {
+    using MaxPolicyT = DispatchFSM::MaxPolicy;
+
+    cudaError_t error;
+
+    // Get PTX version
+    int ptx_version;
+    error = cub::PtxVersion(ptx_version);
+    if (error) return error;
```

Comment:
I prefer to avoid implicit conversion to bool, unless it's preferred in CUB for error checking like this.
Same for the lack of brackets around the branch.
```suggestion
    if (error != cudaSuccess) { return error; }
```